ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/ChainDataPruner.java

```diff
@@ -32,22 +33,30 @@ public class ChainDataPruner implements BlockAddedObserver {
   private final long blocksToRetain;
   private final long pruningFrequency;
   private final ExecutorService pruningExecutor;
+  private final Supplier<Boolean> isInitialSyncPhaseDoneSupplier;
 
   public ChainDataPruner(
       final BlockchainStorage blockchainStorage,
       final ChainDataPrunerStorage prunerStorage,
       final long blocksToRetain,
       final long pruningFrequency,
+      final Supplier<Boolean> isInitialSyncPhaseDoneSupplier,
       final ExecutorService pruningExecutor) {
     this.blockchainStorage = blockchainStorage;
     this.prunerStorage = prunerStorage;
     this.blocksToRetain = blocksToRetain;
     this.pruningFrequency = pruningFrequency;
+    this.isInitialSyncPhaseDoneSupplier = isInitialSyncPhaseDoneSupplier;
     this.pruningExecutor = pruningExecutor;
   }
 
   @Override
   public void onBlockAdded(final BlockAddedEvent event) {
+    if (!isInitialSyncPhaseDoneSupplier.get()) {
+      LOG.debug(
```

Comment:
This could be quite noisy on debug level during the initial sync I think. Maybe we don't need a log here.